chainerx_cc/chainerx/python/routines.cc

```diff
@@ -409,13 +409,86 @@ void InitChainerxManipulation(pybind11::module& m) {
           py::arg("arrays"),
           py::arg("axis") = 0);
     m.def("split",
-          [](const ArrayBodyPtr& ary, int64_t sections, int8_t axis) { return MoveArrayBodies(Split(Array{ary}, sections, axis)); },
-          py::arg("ary"),
-          py::arg("indices_or_sections"),
-          py::arg("axis") = 0);
-    m.def("split",
-          [](const ArrayBodyPtr& ary, std::vector<int64_t> indices, int8_t axis) {
-              return MoveArrayBodies(Split(Array{ary}, indices, axis));
+          [](const ArrayBodyPtr& ary, py::handle indices_or_sections, int8_t axis) {
+              auto split_sections = [](const ArrayBodyPtr& ary, int64_t sections, int8_t axis) {
+                  return MoveArrayBodies(Split(Array{ary}, sections, axis));
+              };
+              auto split_indices = [](const ArrayBodyPtr& ary, const std::vector<int64_t>& indices, int8_t axis) {
+                  return MoveArrayBodies(Split(Array{ary}, indices, axis));
+              };
+
+              // Converts an python float to sections (int64_t).
+              // Raises ValueError if the value has non-zero fraction.
+              auto pyfloat_to_sections_or_value_error = [](py::handle num) {
+                  CHAINERX_ASSERT(py::isinstance<py::float_>(num));
+                  double num_fp = py::cast<double>(num);
+                  auto num_int = static_cast<int64_t>(num_fp);
+                  if (static_cast<double>(num_int) != num_fp) {
+                      throw py::value_error{"Sections must be an integer."};
+                  }
+                  return num_int;
+              };
+
+              // sections: int
+              if (py::isinstance<py::int_>(indices_or_sections)) {
+                  int64_t sections = py::cast<int64_t>(indices_or_sections);
+                  return split_sections(ary, sections, axis);
+              }
+              // sections: float
+              if (py::isinstance<py::float_>(indices_or_sections)) {
+                  int64_t sections = pyfloat_to_sections_or_value_error(indices_or_sections);
+                  return split_sections(ary, sections, axis);
+              }
+              // numpy.ndarray
+              if (py::isinstance<py::array>(indices_or_sections)) {
+                  py::array np = py::cast<py::array>(indices_or_sections);
```

Comment:
I'm not sure about this variable name `np` since the code is quite long and you might further down wonder if it's the numpy module. How about `array` or `np_array`?

Reply:
How about `np_ios`? Both `array` and `np_array` are prone to be confused with `ary`.

Reply:
Maybe that's better, `np_ios` sounds good to me.

Reply:
Fixed.